### ISSUE TITLE
Sudo authorization is discarded between tasks

### ISSUE THREAD

**Issue by jsengla** [ansible/ansible#11345]
The hardcoded "-k" flag to sudo breaks things like One-Time Password authentication, especially since only one password is ever asked for _all_ hosts.

**Comment by bcoca** [comment]
closing as dupe of #9480

**Comment by jsengla** [comment]
As I understand it, SSH is not the same as SUDO...  These are _different_ use cases for OTP, implemented in entirely disparate areas of Ansible's code.
Also, Google Authenticator is not the only 2-factor authentication mechanism in town and is implemented quite differently from the de-facto standard method used for more than a decade via other pam modules.  Google Authenticator prompts for a password (a second one if password authentication is enabled, or a first one if the user is otherwise authenticated, ie, with key exchange).  _Another_ (more common in high security environments) methodology for 2-factor authentication is that the _standard_ SSH (_or_, in the case of this ticket, sudo) password prompt is given just once (thus not necessitating the patch in #9480) and the user has a token to provide the _initial and only_ password which is different every time.  Please see https://en.wikipedia.org/wiki/One-time_password to understand the various technologies.  The patched sshpass to enable Google Authenticator is a special case.
You can consider this feature request a continuation of the long-standing request that Ansible prompts for passwords on a _per-host_ basis for either/both SSH and sudo.  SSH wasn't initially in this ticket, but since you brought it up...
With SSH pipelining enabled, the same authorization can be leveraged to work through a (short) play.  But for sudo, the -k flag is hardcoded to drop credentials on a per-task basis.

**Comment by bcoca** [comment]
Good points all, I made it a duplicate because both cases deal with the same underlying issue, ansible does not currently support OTP either for ssh nor sudo. If we ever find a good way to enable this for ssh, it will be made generic, not just Google and of course we would try to also port it to all become methods (not just sudo). 

I'm currently working on the v2 become authentication, I'll try to remove the -k, which was there to force a password prompt every time as ansible was expecting one and cached credentials broke this. This will not enable OTP but it will be a small step towards it, if I succeed.
